Squads, XO, SO Vendor Organization & QOL

hislittlecuzingames opened the pull request:
# About the pull request

DRAFT UNTIL MADE READY FOR REVIEW. 

Re-organizes how some items appear in Vendors.
Makes some items visible in multiple places. IE: Telephone backpack appears in the backpack section and in the point buying section on some vendors. 

Removes some items from some vendors to reduce bloat.

Makes it so some units like medics can take two belts. 
One section is for their meds, the other is for their gun, removing the 5 slot rifle ammo belt, but allowing SMG and pistol belts which can be placed in armor slots to hold a gun with ammo. 

# Explain why it's good for the game

Some vendors are poorly organized or have a billion items, namely the corpsman vendor. It has basically every pouch available to every unit in the game PLUS medical pouches. 
It's really annoying to navigate/find what you want or figure out what you should take. 

Just holding the players hand with what items should be taken.

Additionally makes some vendors make access to equipment easier as it appears in more locations or in more RP friendly locations. IE Webbing not being in the CIC armory, but being available in the Staff Officer bedrooms. 

# Testing Photographs and Procedure

SEE CODE FOR UP-TO-DATE PHRASING, ORDER, ETC. 

THESE PICTURES ARE ONLY FOR THE INITIAL COMMIT AND TO GIVE IDEAS ON HOW IT MAY APPEAR IN GAME!

<details>
<summary>Screenshots & Videos</summary>
STAFF OFFICER CIC ARMORY LOCKER:
![image](https://github.com/user-attachments/assets/c34fd7cf-4147-48af-ad41-5774fca21c52)

![image](https://github.com/user-attachments/assets/a943250e-a871-444e-8e51-8c596039bb4a)

SQUAD MEDIC LOCKER:
![image](https://github.com/user-attachments/assets/124f661e-f20c-4e31-af59-4d04b4afcc8d)

![image](https://github.com/user-attachments/assets/404e9592-d448-48ba-9597-5a2d1c833593)

![image](https://github.com/user-attachments/assets/d35aa1ea-bc1a-4348-b45f-529def686f97)

Nurse Tools has the surgical kit, pen light, and stethoscope. I consider these items to be clutter as many medics just don't take them. So just QOL of being able to not take them should you choose. 
Or when you want to use them, you can! 

</details>


# Changelog

:cl:
qol: Adjusted Layouts of many Equipment Lockers including XO, SO, SL, FTL, RFN, HM, CT
/:cl:

TheManWithNoHands commented:
Good god. Your mainly disorganizing stuff. Moving magharn out of rail atachments to primary choice is a no go. You dont sort by META/usefullness. EVER. 'cause that stuff changes with every pr making the sorting very quickly invalid.
Sort by names, slots or ANYTHING else.
If you want to let new players know what best to pick, you can highligth options. Dont sort by META though.

Also dont make categorys for each type of belt, thats just clutter and makes it harder to navigate. Otherwise we would have 20 diffrent categorys of stuff. We have belts and pouches for basicly everything, so making categorys for specific belts and pouches makes it only harder to find.

Rough orginisation is important. Go too fine, and you basicly just made one big list. Wich you're aproaching.

Also this needs the balance Tag. Your adding new stuff and removing things. This is not purley QoL.

hislittlecuzingames commented:
Discussion on layout, design, etc continuing here:
https://forum.cm-ss13.com/t/lockers-re-organization/10066

hislittlecuzingames commented:
updated initial comment.
I re-read the code and believe everything I intended to remove I removed & everything I intended to organize was organized.

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

hislittlecuzingames commented:
Sorry Merge conflicts. Why did it change all those other files? All I staged was the one file...

aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

hislittlecuzingames commented:
Stupid Git
